sched/init/nx_start.c

```diff
@@ -534,6 +530,12 @@ void nx_start(void)
 
   g_nx_initstate = OSINIT_TASKLISTS;
 
+  /* Initialize RTOS Data ***************************************************/
+
+  drivers_early_initialize();
```

Comment:
I still feel a little strange about **drivers_early_initialize()**. Driver early initialization is executed before core resource initialization. Does that mean related early initialization such as uart(*****_earlyserialinit()**) could be placed in this implementation?

Reply:
Compared to existing uart(***_earlyserialinit()), they are similar but uart needs to init earlier.

As you have commented out, the sched note facility does need idle tcb to be ready at least.

The bringup process is too compilicated for me to imagine the perfect solution for my coming issue that segger systemview cannot work earlier than memory heap initialization, making it unable to detect correct heap information. Thus this patch comes.

Feel free to make any adjustment to my code as neccesary.

Reply:
I don't have better solution with current implementation, let's keep moving